src/folderlistmodel.cpp

```diff
@@ -402,6 +487,38 @@ void FolderListModel::updateCurrentFolderIndex()
     setCurrentFolderIdx(0);
 }
 
+int FolderListModel::addFolderAndChildren(const QMailFolderId &folderId, QMailMessageKey messageKey,
+                                          QList<QMailFolderId> &originalList, int indexIterator)
+{
+    for (int i = 0; i < originalList.size(); i++) {
+        if (originalList[i] == folderId) {
+            FolderStandardType folderType = folderTypeFromId(originalList[i]);
+            m_folderList.append(createFolderItem(QModelIndex(), originalList[i], folderType,
+                                                 messageKey, indexIterator));
+            indexIterator++;
```

Comment:
'indexIterator' is not an iterator.  In fact, can't you just replace it with 'm_folderList.count()' everywhere?  You probably don't need to pass it into or out of this function, either.